lib/reporters/json.js

```diff
@@ -66,7 +78,13 @@ function JSONReporter(runner, options) {
 
     runner.testResults = obj;
 
-    process.stdout.write(JSON.stringify(obj, null, 2));
+    var json = JSON.stringify(obj, null, 2);
+    if (output) {
+      fs.mkdirSync(path.dirname(output), {recursive: true});
+      fs.writeFileSync(output, json);
```

Comment:
When one of these file operations fails, the error will be swallowed and nothing happens. This is an known problem with Mocha's reporters. So we could use `try/catch` and log the error to the console?

Reply:
done